plugins/net_plugin/net_plugin.cpp

```diff
@@ -1097,32 +1100,38 @@ namespace eosio {
       if( ec ) {
          fc_elog( logger, "connection failed (set_option) ${peer}: ${e1}", ("peer", peer_name())( "e1", ec.message() ) );
          close();
-         return false;
+         callback(false);
```

Comment:
should still return here, after callback is called

Reply:
since method is asynchronous what will be meaning of return?

Reply:
return means nothing later on gets processed.  Not a big deal, but since this is handling an error case early, it makes it clear that nothing later on in the method will be executed.

Reply:
I think it adds additional complexity for caller, as it needs to catch errors in two places (via callback and via return value) so I removed return value. We always execute callback.